velox/dwio/dwrf/writer/FlushPolicy.cpp

```diff
@@ -45,13 +50,14 @@ FlushDecision DefaultFlushPolicy::shouldFlushDictionary(
   if (dictionaryMemoryUsage > dictionarySizeThreshold_) {
     return FlushDecision::FLUSH_DICTIONARY;
   }
-  if (stripeProgress.stripeSizeEstimate >= dictionaryAssessmentThreshold_) {
-    // In the current implementation, since we don't ever change encoding
-    // decision after the first stripe, we don't need to ever reset this
-    // threshold.
-    dictionaryAssessmentThreshold_ +=
-        stripeSizeThreshold_ / kNumDictioanryTestsPerStripe;
-    return FlushDecision::EVALUATE_DICTIONARY;
+  if (stripeIndex_ < stripeProgress.stripeIndex) {
+    dictionaryCheckThreshold_ = 0;
+    setNextDictionaryCheckThreshold();
```

Comment:
```
SCOPE_EXIT{ setNextDictionaryCheckThreshold(stripeProgress.stripeSizeEstimate); };
if (stripeIndex_ < stripeProgress.stripeIndex) {
  setNextDictionaryCheckThreshold();
}
if (stripeProgress.stripeSizeEstimate >= dictionaryCheckThreshold_) {
  return FlushDecision::CHECK_DICTIONARY;
}
```

Reply:
This is more rigorous, but we only need to call it very rarely through the stripe instead of every batch written. I think of the current implementation as lazily incrementing the check threshold.